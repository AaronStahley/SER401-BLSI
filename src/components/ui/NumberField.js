import React from 'react';
import {View, StyleSheet, TextInput} from 'react-native';
import Colors from '../../common/Colors';

export default class NumberField extends React.Component {

    handleChangeText = (value) => {
        if (!isNaN(value)) {
            this.props.onChange((value * 1));
        }
        else if (value === '-') {
            this.props.onChange(value);
        }
        else {
            this.props.onChange(null);
        }
    };

    render() {
        let {prompt, keyboardType, value, onChange} = this.props;
        return (
            <View style={styles.textfieldContainer}>
                <TextInput style={[styles.textfield]}
                           underlineColorAndroid='#0000'
                           placeholder={'please enter a number'}
                           keyboardType={keyboardType}
                           baseColor={Colors.questionBubble}
                           textColor={'#000'}
                           value={!value ? "" : value + ""}
                           onChangeText={this.handleChangeText}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    textfieldContainer   : {
        flex             : 1,
        flexDirection    : "row",
        backgroundColor  : Colors.questionBubble,
        paddingHorizontal: 5,
        paddingTop       : 5,
        paddingBottom    : 5,
        borderColor      : Colors.questionCheckBoxBorder,
        borderWidth      : 1,
        borderRadius     : 2,
        marginHorizontal : 10,
    },
    textfield            : {
        flex             : 1,
        flexDirection    : 'row',
        backgroundColor  : Colors.questionBubble,
        paddingHorizontal: 5,
    }
});