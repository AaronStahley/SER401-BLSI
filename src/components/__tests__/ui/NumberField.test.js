import 'react-native';
import React from 'react';
import NumberField from '../../ui/NumberField';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer
        .create(<NumberField></NumberField>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('handles values correctly', () => {
    const tree = renderer
        .create(<NumberField>123</NumberField>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});