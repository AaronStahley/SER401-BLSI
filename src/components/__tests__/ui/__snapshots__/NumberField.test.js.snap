// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles values correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efefef",
      "borderColor": "#f33338aa",
      "borderRadius": 2,
      "borderWidth": 2,
      "flex": 1,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingBottom": 5,
      "paddingHorizontal": 5,
      "paddingTop": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    baseColor="#efefef"
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#efefef",
          "flex": 1,
          "flexDirection": "row",
          "paddingHorizontal": 5,
        },
      ]
    }
    textColor="#000"
    underlineColorAndroid="#0000"
    value=""
  />
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#efefef",
      "borderColor": "#f33338aa",
      "borderRadius": 2,
      "borderWidth": 2,
      "flex": 1,
      "flexDirection": "row",
      "marginHorizontal": 10,
      "paddingBottom": 5,
      "paddingHorizontal": 5,
      "paddingTop": 5,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    baseColor="#efefef"
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "#efefef",
          "flex": 1,
          "flexDirection": "row",
          "paddingHorizontal": 5,
        },
      ]
    }
    textColor="#000"
    underlineColorAndroid="#0000"
    value=""
  />
</View>
`;
